### PR TITLE
Prevent erroneous using directives from blocking the initial run of BSP

### LINKED ISSUE #1097: Incorrect using directive prevents proper bsp setup
**Version(s)**

0.1.7

**Describe the bug**

When running `scala-cli setup-ide` for the first time in a project, if an incorrect using directive is present (e.g. with missing quotes)  `.scala-build/.bloop` directory and its contents are not generated at all and there's no error about that and then the IDE sees no build targets (and as a consequence the error in the directive is not indicated in the IDE).

**To Reproduce**

```shell
mkdir /tmp/foo
cd /tmp/foo
code .      # open the directory in VSCode
echo '//> using scala 3.1.2' > Foo.scala
scala-cli setup-ide .
```

**Expected behaviour**

Either `scala-cli setup-ide` command should report some error or it should generate the build targets for bsp anyway so that the error is displayed in the IDE.

### PR DESCRIPTION
**Pull request by Gedochao**
Fixes #1097

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/bsp/BspImpl.scala [hunk 1]
```diff
@@ -44,9 +62,19 @@ final class BspImpl(
...
 
+  /** Initial setup for the Bloop project.
+    *
+    * @param currentBloopSession
+    *   the current Bloop session
+    * @param reloadableOptions
+    *   options which may be reloaded on a bsp workspace/reload request
+    * @param isBspInitRun
+    *   a flag indicating if this is the BSP initial setup workflow
+    */
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/bsp/BspImpl.scala]: The name of parameter `isBspInitRun` and description should be more descriptive. 

To understand this flag, I first need to know how BSP initial setup workflow works

File: modules/options/src/main/scala/scala/build/Artifacts.scala [hunk 2]
```diff
@@ -375,12 +380,19 @@ object Artifacts {
...
+    classifiersOpt: Option[Set[String]],
+    isBspInitRun: Boolean = false
   ): Either[BuildException, Fetch.Result] = either {
     val coursierDependenciesWithFallbacks = value {
       dependencies.value
         .map(Positioned(dependencies.positions, _))
         .map(dep => dep.toCs(paramsOpt).map(csDep => (dep.value, csDep.value)))
+        .filter {
+          case Left(l) if isBspInitRun =>
+            logger.error(l.message)
```
Comment (lwronski) [inline modules/options/src/main/scala/scala/build/Artifacts.scala]:
```suggestion
          case Left(e) if isBspInitRun =>
            logger.error(e.message)
```

File: modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala [hunk 3]
```diff
@@ -1118,9 +1120,50 @@ abstract class BspTestDefinitions(val scalaVersionOpt: Option[String])
...
+                new URI(targetUri).getQuery.stripPrefix("id=")
+              }
+
+              val bloopDir = root / Constants.workspaceDirName / ".bloop"
+              expect(os.isDir(bloopDir))
+
+              bloopProjectNames.foreach { bloopProjectName =>
+                val bloopProjectJsonPath = bloopDir / s"$bloopProjectName.json"
+                expect(os.isFile(bloopProjectJsonPath))
+              }
```
Comment (lwronski) [inline modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala]: We should also test, if `bsp` properly generates error diagnostic.

Comment (Gedochao) [reply modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala]: good call, testing for diagnostics made me realise those were broken on the initial run (recovering from errors prevented the silenced errors to be transferred as diagnostics).
ought to be fixed now.

**Review by alexarchambault — COMMENTED** [review]
NIT About the commit structure, so that we have a clean(ish) history on the main branch: could you create a separate commit for the added scaladoc, and squash the other non-refacto commits?

**Comment by Gedochao** [comment]
> NIT About the commit structure, so that we have a clean(ish) history on the main branch: could you create a separate commit for the added scaladoc, and squash the other non-refacto commits?

@alexarchambault should be fixed now.

**Review by lwronski — APPROVED** [review]

**Pull request merged**